lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -592,6 +592,78 @@ static void mergeMatMul(Function *F) {
   }
 }
 
+/// Merge Transpose into MatMul.
```

Comment:
I wonder about the reshape in the optimization. Is the reshape coming from FC node? Would it make sense to catch this pattern before lowering the FC?

Reply:
Great!

Reply:
I realized I was wrong. Reshape comes even before lowering from Caffe2 importer:
https://github.com/pytorch/glow/blob/master/lib/Importer/Caffe2.cpp line 300.
So there is no much sense to move the optimization to pre-lowering stage.